src/h5web/vis-packs/core/line/LineVis.tsx

```diff
@@ -109,7 +106,33 @@ function LineVis(props: Props) {
         }}
       >
         <PanZoomMesh />
-        <TooltipMesh {...tooltipFormatters} guides="vertical" />
+        <TooltipMesh
+          guides="vertical"
+          renderTooltip={(x) => {
+            const xi = abscissaToIndex(x);
+            const abscissa = abscissas[xi];
+
+            if (renderTooltip) {
+              return renderTooltip({ abscissa, xi, x });
+            }
+
+            const value = dataArray.get(xi);
+            if (value === undefined) {
+              return undefined;
+            }
```

Comment:
I believe this piece of code (previously in `useTooltipFormatters`) has been obsolete for a while.

When the user hovers over the extended parts of the live visualization, the `scaleThreshold` (in `getValueToIndexScale`) returns the nearest abscissa index (either 0 or `values.length - 1`). There's no way to disable this automatic "clamping" (since it's not a continuous scale), so we can never get an out-of-range index.

Is that right?

Reply:
If the code is indeed obsolete, then would we still need the ability to "hide" the popup by returning `undefined` in `renderTooltip`?

Reply:
> There's no way to disable this automatic "clamping" (since it's not a continuous scale), so we can never get an out-of-range index.
> 
> Is that right?

Good catch, I think you are right. You can remove this piece code for now and we can think about another way to handle this in another PR (it is not the worst thing to display the first/last value when hovering the extended parts).

Anticipating a bit, I think the best behaviour would be to still display the current abscissa value but do not display the value. We could test that `x` is included in `abscissas` for this ?

Reply:
> Anticipating a bit, I think the best behaviour would be to still display the current abscissa value but do not display the value. We could test that x is included in abscissas for this ?

Yes, it's an easy check. Good idea, it felt wrong hiding the tooltip completely 💯

Reply:
Actually, the `switchAtMidPoints` throws a spanner in the works. If we want to show the first value in the tooltip over the same amount of space as the other values, then we need to know where the mid-point is to the left/bottom of the first abscissa/ordinate value, which isn't straightforward, obviously.

Reply:
I think the behaviour is great as it is. Let's not bother changing it 😄 I'll remove that dead piece of code, though.

Reply:
Then, we could disable the clamping of the scale. But agreed: not necessary to change for now.